File: app/src/main/java/com/example/capstone_donkki/MainFragment.java
package com.example.capstone_donkki;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MainFragment extends Fragment {

    private ArrayList<IngredientData> Ingredient;
    private ArrayList<SearchtagData> searchtag;
    private Ingredient_Adapter ingredient_adapter;
    private Searchtag_Adapter searchtag_adapter;
    private RecyclerView rcv_ingredient, rcv_searchtag;
    private LinearLayoutManager linearLayoutManager;
    private Button btn_search;
    private EditText et_search;
    private TextView tv_refmore;

    private ImageView iv_voice;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View MainView = inflater.inflate(R.layout.frag_main, container, false);

        rcv_ingredient = MainView.findViewById(R.id.rcv_ingredient);
        ingredient_adapter = new Ingredient_Adapter(Ingredient);
        rcv_ingredient.setAdapter(ingredient_adapter);
        rcv_ingredient.setLayoutManager(new LinearLayoutManager(getContext(), RecyclerView.HORIZONTAL, false));

        rcv_searchtag = MainView.findViewById(R.id.rcv_searchtag);
        searchtag_adapter = new Searchtag_Adapter(searchtag);
        rcv_searchtag.setAdapter(searchtag_adapter);

        linearLayoutManager = new LinearLayoutManager(getContext());

        et_search = MainView.findViewById(R.id.et_search);
        btn_search = MainView.findViewById(R.id.btn_search);
        iv_voice = MainView.findViewById(R.id.iv_mic);

        tv_refmore = MainView.findViewById(R.id.tv_refmore);

        return MainView;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        InvokeIngredient();
    }

    private void InvokeIngredient() {
        //DB에서 데이터 가져오기
    }
}

File: app/src/main/java/com/example/capstone_donkki/MainFragment.kt
package com.example.capstone_donkki

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import com.example.capstone_donkki.databinding.FragMainBinding

class MainFragment : Fragment() {
    private lateinit var mainBinding: FragMainBinding
    private var layoutManager: RecyclerView.LayoutManager?=null
    private var adapter: RecyclerView.Adapter<RecyclerAdapter.ViewHolder>?=null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        mainBinding = FragMainBinding.inflate(inflater, container, false)

        return mainBinding.root
    }

    override fun onViewCreated(itemView: View, savedInstanceState: Bundle?) {
        super.onViewCreated(itemView, savedInstanceState)


        //TODO: 프레그먼트 연결
        val IngredientList = arrayListOf(
            //추후 서버 연동 과정 필요
            IngredientData(R.drawable.donkki_half, "식재료명", "0815"),
            IngredientData(R.drawable.donkki_half, "식재료명", "0815")
        )

        val rcv_ingredient: RecyclerView = findViewById(R.id.rcv_ingredient)
        rcv_ingredient.layoutManager =
            LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        rcv_ingredient.setHasFixedSize(true)

        //어뎁터 호출-연결
        rcv_ingredient.adapter = Ingredient_Adapter(IngredientList)

    }
}